File: Projects/Framework-Samples/NHibernate/CSharp/Sample.Data/ManagerObjects/ManagerFactory.cs
using System;
using System.Collections.Generic;
using System.Text;

using Sample.Data.Generated.Base;

namespace Sample.Data.Generated.ManagerObjects
{
    public interface IManagerFactory
    {
		// Get Methods
		IAccountManager GetAccountManager();
		IAccountManager GetAccountManager(INHibernateSession session);
		ICartManager GetCartManager();
		ICartManager GetCartManager(INHibernateSession session);
		ICategoryManager GetCategoryManager();
		ICategoryManager GetCategoryManager(INHibernateSession session);
		IInventoryManager GetInventoryManager();
		IInventoryManager GetInventoryManager(INHibernateSession session);
		IItemManager GetItemManager();
		IItemManager GetItemManager(INHibernateSession session);
		ILineItemManager GetLineItemManager();
		ILineItemManager GetLineItemManager(INHibernateSession session);
		IOrderManager GetOrderManager();
		IOrderManager GetOrderManager(INHibernateSession session);
		IOrderStatusManager GetOrderStatusManager();
		IOrderStatusManager GetOrderStatusManager(INHibernateSession session);
		IProductManager GetProductManager();
		IProductManager GetProductManager(INHibernateSession session);
		IProfileManager GetProfileManager();
		IProfileManager GetProfileManager(INHibernateSession session);
		ISupplierManager GetSupplierManager();
		ISupplierManager GetSupplierManager(INHibernateSession session);
    }

    public class ManagerFactory : IManagerFactory
    {
        #region Constructors

        public ManagerFactory()
        {
        }

        #endregion

        #region Get Methods

		public IAccountManager GetAccountManager()
        {
            return new AccountManager();
        }
		public IAccountManager GetAccountManager(INHibernateSession session)
        {
            return new AccountManager(session);
        }
		public ICartManager GetCartManager()
        {
            return new CartManager();
        }
		public ICartManager GetCartManager(INHibernateSession session)
        {
            return new CartManager(session);
        }
		public ICategoryManager GetCategoryManager()
        {
            return new CategoryManager();
        }
		public ICategoryManager GetCategoryManager(INHibernateSession session)
        {
            return new CategoryManager(session);
        }
		public IInventoryManager GetInventoryManager()
        {
            return new InventoryManager();
        }
		public IInventoryManager GetInventoryManager(INHibernateSession session)
        {
            return new InventoryManager(session);
        }
		public IItemManager GetItemManager()
        {
            return new ItemManager();
        }
		public IItemManager GetItemManager(INHibernateSession session)
        {
            return new ItemManager(session);
        }
		public ILineItemManager GetLineItemManager()
        {
            return new LineItemManager();
        }
		public ILineItemManager GetLineItemManager(INHibernateSession session)
        {
            return new LineItemManager(session);
        }
		public IOrderManager GetOrderManager()
        {
            return new OrderManager();
        }
		public IOrderManager GetOrderManager(INHibernateSession session)
        {
            return new OrderManager(session);
        }
		public IOrderStatusManager GetOrderStatusManager()
        {
            return new OrderStatusManager();
        }
		public IOrderStatusManager GetOrderStatusManager(INHibernateSession session)
        {
            return new OrderStatusManager(session);
        }
		public IProductManager GetProductManager()
        {
            return new ProductManager();
        }
		public IProductManager GetProductManager(INHibernateSession session)
        {
            return new ProductManager(session);
        }
		public IProfileManager GetProfileManager()
        {
            return new ProfileManager();
        }
		public IProfileManager GetProfileManager(INHibernateSession session)
        {
            return new ProfileManager(session);
        }
		public ISupplierManager GetSupplierManager()
        {
            return new SupplierManager();
        }
		public ISupplierManager GetSupplierManager(INHibernateSession session)
        {
            return new SupplierManager(session);
        }
        
        #endregion
    }
}

File: Framework-Samples/NHibernate/VisualBasic/Sample.Data/ManagerObjects/ManagerFactory.vb
Imports System
Imports System.Collections.Generic
Imports System.Text

Imports Sample.Data.Generated.Base

Namespace Sample.Data.Generated.ManagerObjects
	Public Interface IManagerFactory
		' Get Methods
		Function GetAccountManager() As IAccountManager
		Function GetAccountManager(ByVal session As INHibernateSession) As IAccountManager
		Function GetCartManager() As ICartManager
		Function GetCartManager(ByVal session As INHibernateSession) As ICartManager
		Function GetCategoryManager() As ICategoryManager
		Function GetCategoryManager(ByVal session As INHibernateSession) As ICategoryManager
		Function GetInventoryManager() As IInventoryManager
		Function GetInventoryManager(ByVal session As INHibernateSession) As IInventoryManager
		Function GetItemManager() As IItemManager
		Function GetItemManager(ByVal session As INHibernateSession) As IItemManager
		Function GetLineItemManager() As ILineItemManager
		Function GetLineItemManager(ByVal session As INHibernateSession) As ILineItemManager
		Function GetOrderManager() As IOrderManager
		Function GetOrderManager(ByVal session As INHibernateSession) As IOrderManager
		Function GetOrderStatusManager() As IOrderStatusManager
		Function GetOrderStatusManager(ByVal session As INHibernateSession) As IOrderStatusManager
		Function GetProductManager() As IProductManager
		Function GetProductManager(ByVal session As INHibernateSession) As IProductManager
		Function GetProfileManager() As IProfileManager
		Function GetProfileManager(ByVal session As INHibernateSession) As IProfileManager
		Function GetSupplierManager() As ISupplierManager
		Function GetSupplierManager(ByVal session As INHibernateSession) As ISupplierManager
	End Interface

	Public Class ManagerFactory
		Implements IManagerFactory
		
#region "Constructors"
		Public Sub New()
		End Sub
#End Region

#region "Get Methods"

		Public Function GetAccountManager() As IAccountManager Implements IManagerFactory.GetAccountManager
			Return New AccountManager()
		End Function
		Public Function GetAccountManager(ByVal session As INHibernateSession) As IAccountManager Implements IManagerFactory.GetAccountManager
			Return New AccountManager(session)
		End Function
		Public Function GetCartManager() As ICartManager Implements IManagerFactory.GetCartManager
			Return New CartManager()
		End Function
		Public Function GetCartManager(ByVal session As INHibernateSession) As ICartManager Implements IManagerFactory.GetCartManager
			Return New CartManager(session)
		End Function
		Public Function GetCategoryManager() As ICategoryManager Implements IManagerFactory.GetCategoryManager
			Return New CategoryManager()
		End Function
		Public Function GetCategoryManager(ByVal session As INHibernateSession) As ICategoryManager Implements IManagerFactory.GetCategoryManager
			Return New CategoryManager(session)
		End Function
		Public Function GetInventoryManager() As IInventoryManager Implements IManagerFactory.GetInventoryManager
			Return New InventoryManager()
		End Function
		Public Function GetInventoryManager(ByVal session As INHibernateSession) As IInventoryManager Implements IManagerFactory.GetInventoryManager
			Return New InventoryManager(session)
		End Function
		Public Function GetItemManager() As IItemManager Implements IManagerFactory.GetItemManager
			Return New ItemManager()
		End Function
		Public Function GetItemManager(ByVal session As INHibernateSession) As IItemManager Implements IManagerFactory.GetItemManager
			Return New ItemManager(session)
		End Function
		Public Function GetLineItemManager() As ILineItemManager Implements IManagerFactory.GetLineItemManager
			Return New LineItemManager()
		End Function
		Public Function GetLineItemManager(ByVal session As INHibernateSession) As ILineItemManager Implements IManagerFactory.GetLineItemManager
			Return New LineItemManager(session)
		End Function
		Public Function GetOrderManager() As IOrderManager Implements IManagerFactory.GetOrderManager
			Return New OrderManager()
		End Function
		Public Function GetOrderManager(ByVal session As INHibernateSession) As IOrderManager Implements IManagerFactory.GetOrderManager
			Return New OrderManager(session)
		End Function
		Public Function GetOrderStatusManager() As IOrderStatusManager Implements IManagerFactory.GetOrderStatusManager
			Return New OrderStatusManager()
		End Function
		Public Function GetOrderStatusManager(ByVal session As INHibernateSession) As IOrderStatusManager Implements IManagerFactory.GetOrderStatusManager
			Return New OrderStatusManager(session)
		End Function
		Public Function GetProductManager() As IProductManager Implements IManagerFactory.GetProductManager
			Return New ProductManager()
		End Function
		Public Function GetProductManager(ByVal session As INHibernateSession) As IProductManager Implements IManagerFactory.GetProductManager
			Return New ProductManager(session)
		End Function
		Public Function GetProfileManager() As IProfileManager Implements IManagerFactory.GetProfileManager
			Return New ProfileManager()
		End Function
		Public Function GetProfileManager(ByVal session As INHibernateSession) As IProfileManager Implements IManagerFactory.GetProfileManager
			Return New ProfileManager(session)
		End Function
		Public Function GetSupplierManager() As ISupplierManager Implements IManagerFactory.GetSupplierManager
			Return New SupplierManager()
		End Function
		Public Function GetSupplierManager(ByVal session As INHibernateSession) As ISupplierManager Implements IManagerFactory.GetSupplierManager
			Return New SupplierManager(session)
		End Function
		
#End Region
	End Class
End Namespace
